### PR TITLE
fix(popover): close popover on escape

### PR DESCRIPTION
**Pull request by jinlee93**
### Summary:
- now closes popover on escape
  - tested in unit test
- also includes popover unit test snaps
### Test Plan:
- tests pass
- no visual regressions

[sc-201722]

### CODE REVIEW

File: src/components/Popover/Popover.test.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    render(<Default />);
+    userEvent.click(screen.getByRole('button'));
+    expect(screen.getByTestId('popover-test')).toBeInTheDocument();
+    expect(screen.getByTestId('popover-test')).toHaveAttribute(
+      'aria-hidden',
+      'false',
+    );
+    // Needed to focus inside the popover due to animation
+    userEvent.tab();
+    expect(screen.getByTestId('popover-test')).toHaveFocus();
```
Comment (jinlee93) [inline src/components/Popover/Popover.test.tsx:29]: Alternative could have `screen.getByTestId('popover-test').focus()` but I felt using `tab()` felt more user reflective

Comment (jinlee93) [reply src/components/Popover/Popover.test.tsx]: Fixed with `isActive` being `true` having Popover open to begin with

File: src/components/Popover/Popover.tsx [hunk 2]
```diff
@@ -154,8 +154,8 @@ export const Popover = ({
    * Handle onKeyDown
    * 1) If escape button is struck, close the popover
```
Comment (dierat) [inline src/components/Popover/Popover.tsx:155]: "is struck" huh, never hear it described quite like that

File: src/components/Popover/Popover.test.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+const { Default } = composeStories(PopoverStoryFile);
+PopoverStoryFile.default.args = {
+  isActive: true,
+};
+
+describe('<Popover />', () => {
+  generateSnapshots(PopoverStoryFile);
+
+  it('should close Popover via escape key', async () => {
```
Comment (dierat) [inline src/components/Popover/Popover.test.tsx:16]: Nice test, thanks!

**Review by dierat — APPROVED** [review]

**Pull request merged**